src/synology_dsm/api/surveillance_station/camera.py

```diff
@@ -66,35 +69,44 @@ def is_recording(self):
 class SynoCameraLiveView:
     """An representation of a Synology SurveillanceStation camera live view."""
 
-    def __init__(self, data):
+    def __init__(self, data, base_url):
         """Initialize a Surveillance Station camera live view."""
+        self._base_url = base_url
         self.update(data)
 
+    def __format(self, url):
+        """Format live stream based on base url."""
+        parsed_url = urlparse(url)
+        new_url = parsed_url._replace(
+            netloc=self._base_url.join(parsed_url.netloc.rsplit(parsed_url.hostname, 1))
```

Comment:
That's not really readable. Can you please try to create intermediate variables.